states/huntsman/twilight_flat_fielding.py

```diff
@@ -0,0 +1,71 @@
+"""
+State to handle the taking of calibration frames (evening and morning).
+"""
+
+
+def _get_cameras(pocs):
+    '''
+    Get lists of narrow and broad band cameras.
+    '''
+    narrow_band_cameras = list()
+    broad_band_cameras = list()
+    for cam_name, cam in pocs.observatory.cameras.items():
+
+        # This is a hack. There should be a narrow property in the config...
+        if cam.filter_type.lower().startswith('ha'):
+            narrow_band_cameras.append(cam_name)
+        else:
+            broad_band_cameras.append(cam_name)
+
+    return narrow_band_cameras, broad_band_cameras
+
+
+def _wait_for_twilight(pocs):
+    '''
+    Wait for twilight. Temporary solution until something better is found.
+    '''
+    pocs.logger.debug('Waiting for twilight...')
+    while pocs.is_dark(horizon='focus') or not pocs.is_safe(horizon='flat'):
+        pocs.sleep(delay=pocs._safe_delay)
```

Comment:
> I don't know if the weather system sends messages if the weather turns bad. @wtgee?

No, but ideally it should.

We use the `wait_until_safe` to handle just this kind of situation but it's not clear to me that will work here since what we want to wait on is a specific darkness level and not an overall safety.

@AnthonyHorton suggestion is okay but doesn't check things like AC and disk space.

Reply:
I think I had something similar in my original implementation. But yes, you are right!

Reply:
@wtgee Yep, some awkwardness comes from the fact that `is_safe()` is getting used for actual safety decisions (i.e. need to shut down because it's raining) and also for deciding whether it's the right time to do something (i.e. not too bright/not too dark, and also safe). Could do with either an additional method or an option for `is_safe()` so that we can easily just check weather, AC power and disk space independently of Sun altitude. Or maybe just use `is_safe()` as is, but with a horizon of 0 degrees Sun altitude, because running when the Sun is up is a genuine safety risk?

Reply:
I think we can use `while pocs.is_safe(horizon='flat'):` here, because `pocs.is_dark(horizon='flat')` is required by the state machine to enter the state in the first place. Then we can just use `twilight = not pocs.is_dark(horizon='focus')` within the while loop.